presentation/matrix_view.py

```diff
@@ -46,7 +46,8 @@ def __init__(self, config: Config) -> None:
 
     def make_layout(self, mesh: MeshResults, metric: MetricType, config: Config) -> html.Div:
         self._agents = mesh.agents  # remember agents used to make the layout for further processing
-        timestampISO = mesh.utc_timestamp.isoformat()
+        timestamp_low_ISO = mesh.utc_timestamp_low.isoformat() if mesh.utc_timestamp_low else "none"
```

Comment:
I wonder under which circumstances those timestamps will be missing while there are any data in the matrix (assuming the `has_data` method is corrected).

Reply:
Matrix can be empty when there is no health data available for requested time window. Then, the timestamp will be None

Reply:
You're right.